src/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,75 @@
+import { ChangeEventHandler, ComponentProps, FC, ReactNode } from 'react';
+
+import Icon from '../Icon';
+import Text from '../Text';
+
+import CheckboxIcon from './CheckboxIcon';
+import CheckboxInput from './CheckboxInput';
+import CheckboxLabel from './CheckboxLabel';
+import StyledCheckbox from './StyledCheckbox';
+
+type CheckboxLabelProps = ComponentProps<typeof CheckboxLabel>;
+
+type Props = CheckboxLabelProps & {
+  checked?: boolean;
+  disabled?: boolean;
+  icon?: string;
+  id: string;
+  indeterminate?: boolean;
+  isInvalid?: boolean;
+  label: ReactNode;
+  name: string;
+  onChange?: ChangeEventHandler<HTMLInputElement>;
+  value?: string;
+};
+
+const Checkbox: FC<Props> = ({
+  checked = false,
+  disabled = false,
+  icon,
+  id,
+  indeterminate = false,
+  isInvalid = false,
+  label,
+  name,
+  onChange = () => {},
```

Comment:
There's a warning if `checked` is supplied to an `input` with no `onChange`, so we pass a no-op by default.